istioctl/cmd/analyze.go

```diff
@@ -172,13 +193,30 @@ istioctl experimental analyze -k -d false
 				}
 			}
 
-			// Print validation message output, or a line indicating that none were found
-			if len(outputMessages) == 0 {
-				fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
-			} else {
-				for _, m := range outputMessages {
-					fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
+			switch msgOutputFormat {
+			case LogOutput:
+				// Print validation message output, or a line indicating that none were found
+				if len(outputMessages) == 0 {
+					fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
+				} else {
+					for _, m := range outputMessages {
+						fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
```

Comment:
We can have this print take place outside the switch statement, to avoid duplicating it.

Reply:
I think this just adds code without making it easier to understand. We'll still need a line in each case block to set an output string instead of the print, and then an additional line outside of the switch statement to then print it. We'd also need another variable to hold whether we're printing to stderr or stdout since it's different.

I think the repeated printing is better handled by not using a switch statement, but like I mentioned in the other comment I'd like to put it into a different PR and do proxyconfig at the same time so we keep the style consistent.